docs/extras.md

```diff
@@ -1,11 +1,108 @@
 # Extras
 
+Aside from the [three main API functions](/api), `hermes` also exports these extras.  Some are used internally and exposed just-in-case, while others you my find useful all the time.  It's up to you.
+
 ## getCategoryMessages
 
+```haskell
+getCategoryMessages :: Object -> Stream Message
+```
+
+Accepts an object of options and returns a [Highland Stream](http://highlandjs.org/) of messages.  Used internally by [consumers](/api?id=consumer) to poll the message store for new messages in a category.
+
+| Option        | Type     | Default | Description                                                                                         |
+|:--------------|:---------|:--------|:----------------------------------------------------------------------------------------------------|
+| `batchSize`   | `Number` | `1000`  | Number of messages to retrieve in a single batch                                                    |
+| `category`    | `String` |         | Category name from which to retrieve messages (required)                                            |
+| `condition`   | `String` |         | SQL condition to filter the batch further                                                           |
+| `correlation` | `String` |         | Filters the batch by a category or stream name recorded as `message.metadata.correlationStreamName` |
+| `groupMember` | `Number` |         | Zero-based member number of an individual consumer in a consumer group                              |
+| `groupSize`   | `Number` |         | Total size of a consumer group                                                                      |
+| `position`    | `Number` | `1`     | Global position to start from                                                                       |
+
+```js
+const { getCategoryMessages } = require('../lib/hermes')
+
+getCategoryMessages({
+  category: `userActivation`,
+  position: 100
+}).flatMap(handleMessage)
+  .done(() => console.log('done'))
+```
+
 ## getLastStreamMessage
 
+```haskell
+getLastStreamMessage :: String -> Promise Message
+```
+
+Accepts a stream name, and resolves with the last message in the stream if available, or `undefined` if the stream is empty.  Used internally by [consumers](/api?id=consumer) to load their last known position, and by [entity stores](/api?id=entity) to [load snapshots](/api?id=snapshots).
+
+```js
+const { getLastStreamMessage } = require('../lib/hermes')
+
+getLastStreamMessage(`userActivation-${userId}`).then(console.log)
+```
+
 ## getStreamMessages
 
+```haskell
+getStreamMessages :: Object -> Stream Message
+```
+
+Accepts an object of options, and returns a [Highland Stream](http://highlandjs.org/) of messages.  Used internally by [entity stores](/api?id=entity) to project over the messages in an individual stream.
```

Comment:
Somehow i missed this was returning a highland stream. Any reason not to return a `Readable`?

Reply:
Mostly for convenience.  It's much simpler to `.map()` over a Highland stream to parse the messages than to build a transform stream, and I'd like it in a Highland stream anyways when it's used internally.  `Entity` does a `.reduce()` over the result of `getStreamMessages`, and `Consumer` does a `.flatMap()` and `.stopOnError()` over the result of `getCategoryMessages`.  That's all so much simpler to implement with Highland than a `Readable`.

Reply:
I have no objection to using highland internally. Externally, though, i feel it may be better to transform the stream into a `Readable` as not to lock the consuming code into a specific vendor. 

Not a hill i want do die on, though. If we return a `Readable`, i'll probably just wrap it in a highland stream anyway. Though who knows; maybe i'll want to use rxjs one day...

Reply:
How about if I add a note:
> If you'd like a regular Node Readable stream instead, then call [.toNodeStream(opts)](http://highlandjs.org/#toNodeStream) on the result.

Reply:
:+1: `.toNodeStream({ objectMode: true })`. I always forget that last part.